solarforecastarbiter/reports/template.py

```diff
@@ -132,6 +132,16 @@ def _figure_name_filter(value):
     return out
 
 
+def _unique_flags_filter(proc_fxobs_list):
+    # use a dict to preserve order
```

Comment:
is preserving order from the input important? I think the order should make the validation table look like

![image](https://user-images.githubusercontent.com/4625457/97469687-ed31eb00-1903-11eb-8d6b-24ca79a7f821.png)

Reply:
maybe this requires matching to the quality_flags to find discard_before_resample, so not trivial

Reply:
Flags like `DISCARD BEFORE RESAMPLE: USER FLAGGED OR ISNAN` describe resampled intervals that were discarded because too many of the raw observations were discared within that longer interval. I admit that is not clear but I'm not sure what to do about it. Ideas?

Reply:
Split the table. First table has counts of flagged points before resampling, second is of resampled counts discarded b/c of flags. I'm not sure how to make this work for older reports though

![image](https://user-images.githubusercontent.com/4625457/97472879-8f9f9d80-1907-11eb-94fa-f3916024fdcf.png)

Reply:
Ah, in rc3, validation result is always the counts before resampling. So, I think we can add a field to ValidationResult `before_resample: bool = True`. Then make that first table for all results with `before_resample == True` and the second for `before_resample == False`. Maybe put an alert to update the report or something if there are no ValidationResult w/ `before_resample == False`. Then we can also get rid of that DISCARD BEFORE RESAMPLE prefix

Reply:
The `filter_resample` function could prepend `'_resample '` to the relevant flags. Then the templating process could filter on that and remove the text from the printed name. I think that's backwards compatible since the default is everything goes into discard before resample. The total keys just would be printed because they don't exist in the old reports.

Alternatively, the `ValidationResult` could get a new key `discard_before_resample` (maybe even `resample_threshold_percentage` too). I'm guessing it would be backwards compatible if they had defaults that matched the existing `QualityFlagFilter` defaults. `filter_resample` could return a list of `ValidationResults` instead of a dict of counts, or the counts dict could be modified. I think we discussed similar ideas when considering #580.

Reply:
ha, so I guess it's decided.

Reply:
html is working here https://github.com/wholmgren/solarforecastarbiter-core/tree/valresultrefactor